src/sql/run/resultset.py

```diff
@@ -420,7 +421,16 @@ def fetchmany(self, size):
             # psycopg2 raises psycopg2.ProgrammingError error when running a script
             # that doesn't return rows e.g, 'CREATE TABLE' but others don't
             # (e.g., duckdb), so here we catch all
-            except Exception:
+            except Exception as e:
+                if not any(
+                    substring in str(e)
+                    for substring in [
+                        "This result object does not return rows",
+                        "no results to fetch",
```

Comment:
what are the error types of these two error messages? cause I'm thinking that it might be better to distinguish them by type as well

but if they're generic, this substring solution might be the best one

Reply:
ok, I think let's leave the current solution as is. I see you're still making changes. let me know once this is ready

Reply:
It's ready. 
I was just adding the type check also but then reverted the changes